File: core/src/org/rexcellentgames/burningknight/entity/item/weapon/gun/shotgun/Shotgun.java
package org.rexcellentgames.burningknight.entity.item.weapon.gun.shotgun;

import org.rexcellentgames.burningknight.entity.item.weapon.gun.Gun;
import org.rexcellentgames.burningknight.util.Random;
import org.rexcellentgames.burningknight.util.geometry.Point;

public class Shotgun extends Gun {
	{
		accuracy = 20;
		useTime = 3f;
		damage = 3;
	}

	@Override
	protected void sendBullets() {
		Point aim = this.owner.getAim();
		float an = this.owner.getAngleTo(aim.x, aim.y);

		for (int i = 0; i < 3; i++) {
			float a = (float) (an - Math.PI * 2);
			this.sendBullet((float) (a + Math.toRadians(Random.newFloat(-this.accuracy, this.accuracy))));
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/weapon/gun/shotgun/Shotgun.kt
package org.rexcellentgames.burningknight.entity.item.weapon.gun.shotgun

import org.rexcellentgames.burningknight.entity.item.weapon.gun.Gun
import org.rexcellentgames.burningknight.util.Random

open class Shotgun : Gun() {
  init {
    accuracy = 20f
    useTime = 3f
    damage = 3
    sprite = "item-gun_a"
  }

  override fun sendBullets() {
    val aim = this.owner.aim
    val an = this.owner.getAngleTo(aim.x, aim.y)
    val a = (an - Math.PI * 2).toFloat()

    for (i in 0 .. 4) {
      this.sendBullet((a + Math.toRadians(Random.newFloat(-this.accuracy, this.accuracy).toDouble())).toFloat())
    }
  }
}